app/services/gmail/download.py

```
@@ -170,28 +169,31 @@ def callback(
-        state.download_status["message"] = f"Ready to download {len(email_data)} emails"
+        state.update_download_status(
+            csv_data=output.getvalue(),
+            progress=100,
+            done=True,
+            message=f"Ready to download {len(email_data)} emails",
+        )
 
     except Exception as e:
-        state.download_status["done"] = True
-        state.download_status["error"] = f"Error generating CSV: {str(e)}"
+        state.update_download_status(done=True, error=f"Error generating CSV: {str(e)}")
```

Ruff (0.14.10)

Use explicit conversion flag

Replace with conversion flag

(RUF010)